.github/workflows/build.yml

```diff
@@ -0,0 +1,109 @@
+name: Build
+
+on:
+    push:
+        branches: [ master ]
+    pull_request:
+    release:
+        types: [ created ]
+
+jobs:
+    tests:
+        runs-on: ubuntu-latest
+        name: Build and test
+        strategy:
+            fail-fast: false
+            matrix:
+                include:
+                    -   php: 7.2
+                        tools: "composer:v1"
+                        git-fetch-depth: 1
+                    -   php: 7.3
+                        tools: "composer:v1"
+                        publish-phar: true
+                        git-fetch-depth: 0 # box/composer needs history to determine a recent git version
+                    -   php: 7.3
+                        tools: "composer:v1"
+                        phar-readonly: true
+                        git-fetch-depth: 1
+                    -   php: 7.4
+                        tools: "composer:v1"
+                        coverage: "pcov"
+                        git-fetch-depth: 1
+                    -   php: 7.4
+                        tools: "composer:v2"
+                        git-fetch-depth: 1
+
+        steps:
+            -   uses: actions/checkout@v2
+                with:
+                    fetch-depth: "${{ matrix.git-fetch-depth }}"
+
+            -   name: Setup PHP
+                uses: shivammathur/setup-php@v2
+                with:
+                    php-version: "${{ matrix.php }}"
+                    ini-values: "phar.readonly=0"
+                    tools: "${{ matrix.tools }}"
+                    coverage: "${{ matrix.coverage }}"
+                    extensions: ctype, iconv, xml
+
+            -   name: Get composer cache directory
+                id: composercache
+                run: echo "::set-output name=dir::$(composer config cache-files-dir)"
+
+            -   name: Cache composer dependencies
+                uses: actions/cache@v2
+                with:
+                    path: ${{ steps.composercache.outputs.dir }}
+                    key: ${{ runner.os }}-composer-cache-${{ hashFiles('**/composer.lock') }}
+                    restore-keys: ${{ runner.os }}-composer-cache-
```

Comment:
Both key and restore-keys should include `${{ matrix.php }}`, and restore keys could use two rows with and without  PHP version.

Reply:
You can see here how it's done: https://github.com/infection/infection/blob/b8385ca8960488759f0e4cb625f0f9ab222a10d7/.github/workflows/ci.yaml#L50-L56

Reply:
@sanmai Thank you for your review. I think sometimes it would be more helpful and politely if there is more about why it should be changed instead of how it must be. First I felt a bit under attack but I guess it wasn't meant to be.

The advantage about multiple restore keys would be if dependencies got updated the job won't have an empty cache but use the cache of the job that run before, right?

Reply:
@smoench I'm really sorry about that. I wasn't meaning an attack or anything like that.

That's right, and not only. Caches might be purged by GitHub from time to time. But it really can help to have good granularity for the initial key. Say, if you don't include a PHP version, then a job might have to download some one-off dependency every single time because caches are not refreshed.

Reply:
Thanks for the clarification. I adjusted it.